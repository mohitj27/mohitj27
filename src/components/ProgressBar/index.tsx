/**
 * ProgressBar component.
 */
import React from 'react';
import Color from 'color';
import { bgMidBlue, green } from '../../theme';

export interface ProgressBarProps {
  value: number;
  color?: string;
  bgColor?: string;
  width?: number;
  height?: number;
}

const ProgressBar: React.FC<ProgressBarProps> = ({
  value,
  color = green,
  bgColor = Color(bgMidBlue).lighten(0.125).string(),
  width = 500,
  height = 3,
}) => {
  const containerStyles = {
    width,
    height,
    backgroundColor: bgColor,
    marginTop: '3px',
  };

  const fillerStyles = {
    height: '100%',
    width: `${value}%`,
    backgroundColor: color,
    transition: 'width 500ms ease',
  };

  return (
    <div style={containerStyles}>
      <div style={fillerStyles}></div>
    </div>
  );
};

export default ProgressBar;
