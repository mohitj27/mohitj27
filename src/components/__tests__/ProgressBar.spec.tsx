import React from 'react';
import { render } from '@testing-library/react';
import ProgressBar from '../ProgressBar';

describe('<ProgressBar />', () => {
  test('should render correctly', () => {
    const { asFragment } = render(<ProgressBar value={80} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
