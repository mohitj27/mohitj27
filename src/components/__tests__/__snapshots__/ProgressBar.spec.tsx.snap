// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProgressBar /> should render correctly 1`] = `
<DocumentFragment>
  <div
    style="width: 500px; height: 3px; background-color: hsl(220, 13%, 20%); margin-top: 3px;"
  >
    <div
      style="height: 100%; width: 80%; background-color: rgb(152, 195, 121); transition: width 500ms ease;"
    />
  </div>
</DocumentFragment>
`;
